app/crud/user_management.py

```diff
@@ -0,0 +1,113 @@
+from sqlalchemy import text
+from sqlalchemy.orm import Session
+from app.models.user_models import User 
+from fastapi import HTTPException
+
+def create_user(db:Session,name:str,email:str,hashed_password:str):
+    """
+        Create a new user in the database.
+
+        This function checks if a user with the given email already exists. If not, it hashes
+        the provided plaintext password and creates a new user with the provided name, email,
+        and hashed password.
+
+        Parameters:
+            - db (Session): The database session.
+            - name (str): The name of the user.
+            - email (str): The email of the user.
+            - password (str): The plaintext password of the user.
+
+        Raises:
+            - HTTPException: If the email already exists (status code 400).
+            - HTTPException: If there is an error creating the user (status code 500).
+
+        Returns:
+            - User: The newly created user object.
+    """
+    try:
+        existing_user= get_user_by_email(db,email)
+        if existing_user:
+            raise HTTPException(status_code=400,detail="Email already exist") 
+        new_user = User(name=name,email=email,hashed_password=hashed_password)
+        db.add(new_user)
+        db.commit()
+        db.refresh(new_user)
+        return new_user
+    except Exception as e:
+        db.rollback()
+        raise HTTPException(status_code=500,detail=f"Error on creating user: {e}")
```

Comment:
Its always a best practice to logging full exception stack trace for deeper debugging.

also log important IO events as most of the time they help debug your code. Effective logging save countless hours of debugging.

Reply:
I have completed all the tasks as per the changes you suggested, sir.